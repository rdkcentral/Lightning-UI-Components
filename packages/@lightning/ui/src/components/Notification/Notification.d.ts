import type lng from '@lightningjs/core';
import type {
  Base,
  Color,
  StylePartial,
  TextBoxStyles
} from '@lightning/ui-core';

type Animation = {
  duration?: number;
  delay?: number;
};

type LogoStylesObject = {
  radius: lng.Tools.CornerRadius;
  h: number;
  w: number;
};

export type NotificationStyles = {
  w: number;
  backgroundColor: Color;
  iconStyles: Record<string, unknown>;
  logoStyles: LogoStylesObject;
  iconOnlyMarginX: number;
  marginX: number;
  marginY: number;
  iconSize: number;
  iconMarginRight: number;
  iconMarginTop: number;
  titleTextStyle: TextBoxStyles;
  descriptionTextStyle: TextBoxStyles;
  actionAreaMarginTop: number;
  actionAreaTextStyle: TextBoxStyles;
  enterAlpha: number;
  enterScale: number;
  dismissAlpha: number;
  dismissScale: number;
  step1Animation: Animation;
  step2Animation: Animation;
  step2TextEnterAnimation: Animation;
  step2TextExitAnimation: Animation;
  step3Animation: Animation;
};

export default class Notification extends Base {
  icon?: string;
  logo?: string;
  title?: string;
  description?: string;
  actionArea?: string;
  entered?: boolean;
  get style(): NotificationStyles;
  set style(v: StylePartial<NotificationStyles>);

  // tags
  get _Container(): lng.Component;
  get _Content(): lng.Component;
  get _Icon(): lng.Component;
  get _Text(): lng.Component;
  get _Title(): lng.Component;
  get _Description(): lng.Component;
  get _ActionArea(): lng.Component;
  get _ActionAreaText(): lng.Component;
}
