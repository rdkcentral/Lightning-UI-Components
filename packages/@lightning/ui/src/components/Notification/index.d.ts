import Notification, { NotificationStyle } from './Notification';

export { Notification as default, NotificationStyle };
