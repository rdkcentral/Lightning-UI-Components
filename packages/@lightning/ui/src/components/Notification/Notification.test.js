import TestUtils from '@lightning/ui-core/test';
import defaultIcon from '../../assets/images/notification_64.png';
import { jest } from '@jest/globals';
import Notification from '.';

const createNotification = TestUtils.makeCreateComponent(Notification);

describe('Notification', () => {
  let notification, testRenderer;

  beforeEach(() => {
    [notification, testRenderer] = createNotification({
      title: 'Title',
      description: 'This is a description',
      icon: 'patch/to/src'
    });
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders an inactive Notification', () => {
    const tree = testRenderer.toJSON(2);
    expect(tree).toMatchSnapshot();
  });

  it('renders an active Notification', async () => {
    [notification, testRenderer] = createNotification(
      {
        title: 'Title',
        description: 'Description',
        actionArea: 'action area',
        entered: true
      },
      { spyOnMethods: ['enter'] }
    );

    await notification._enterSpyPromise;
    TestUtils.fastForward([
      notification._Container,
      notification._Icon,
      notification._Text
    ]);
    testRenderer.update();

    const tree = testRenderer.toJSON(2);
    expect(tree).toMatchSnapshot();
  });

  it('sets an icon', () => {
    expect(notification._Icon.icon).toEqual('patch/to/src');
  });

  it('should set a logo', () => {
    const logoPath = 'path/to/logo';
    notification.logo = logoPath;
    testRenderer.forceAllUpdates();

    expect(notification._Icon.icon).toBe(logoPath);
    expect(notification._Icon.style.color).toBeUndefined();
  });

  it('should use the default icon when not icon or logo is set', () => {
    notification.icon = undefined;
    testRenderer.forceAllUpdates();

    expect(notification._Icon.icon).toBe(defaultIcon);
  });

  it('should center a title in the notification when there is no description', () => {
    const title = 'title';
    [notification] = createNotification({ title });
    expect(notification._Title.content).toEqual(title);
    expect(notification._Content.flex).toMatchObject({
      direction: 'row',
      alignItems: 'center'
    });
  });

  it('will not show the action area if it is undefined', () => {
    [notification, testRenderer] = createNotification({
      title: 'Title',
      description: 'This is a description',
      icon: 'patch/to/src'
    });
    expect(notification._ActionArea.alpha).toBe(0.001);
  });

  it('will show the action area if one is passed in', async () => {
    [notification, testRenderer] = createNotification({
      actionArea: 'Action Area'
    });
    expect(notification._ActionArea.alpha).toBe(0.001);

    await notification.enter();

    expect(notification._ActionArea.alpha).toBe(1);
    expect(notification._ActionAreaText.content).toBe('Action Area');
  });

  it('should display a notification without a title or description', () => {
    [notification, testRenderer] = createNotification({ icon: 'patch/to/src' });
    expect(notification._Title.content).toBeUndefined();
    expect(notification._Description.content).toBeUndefined();
  });

  it('should be able to mount entered', async () => {
    [notification, testRenderer] = createNotification(
      { entered: true },
      { spyOnMethods: ['enter'] }
    );

    await notification._enterSpyPromise;
    TestUtils.fastForward([
      notification._Container,
      notification._Icon,
      notification._Text
    ]);
    testRenderer.update();

    expect(notification._Container.alpha).toEqual(
      notification.style.enterAlpha
    );
    expect(notification._Container.scale).toEqual(
      notification.style.enterScale
    );
    expect(notification._Icon.alpha).toEqual(notification.style.enterAlpha);
    expect(notification._Icon.scale).toEqual(notification.style.enterScale);
    expect(notification._Text.alpha).toEqual(notification.style.enterAlpha);
  });

  it('should center content text if description or title is removed after the Notification is activated', async () => {
    [notification, testRenderer] = createNotification(
      {
        entered: true,
        title: 'Title',
        description: 'description',
        actionArea: 'Action Area'
      },
      { spyOnMethods: ['enter', '_update', '_updateLayout'] }
    );

    await notification.__updateSpyPromise;
    await notification._enterSpyPromise;
    TestUtils.fastForward([
      notification._Container,
      notification._Icon,
      notification._Text
    ]);
    testRenderer.update();

    notification.description = '';
    await notification.__updateSpyPromise;

    // assert changes from _updateLayout happened
    await notification._updateLayoutSpyPromise;
    expect(notification._Content.flex.alignItems).toBe('center');

    notification.title = '';
    notification.description = 'description';
    await notification.__updateSpyPromise;

    // assert changes from _updateLayout happened
    await notification._updateLayoutSpyPromise;
    expect(notification._Content.flex.alignItems).toBe('center');
  });

  it('should update the layout if changes are made while the Notification is activated', async () => {
    [notification, testRenderer] = createNotification(
      {
        entered: true,
        title: 'Title',
        description: 'description',
        actionArea: 'Action Area'
      },
      { spyOnMethods: ['enter', '_updateLayout'] }
    );

    await notification._enterSpyPromise;
    TestUtils.fastForward([
      notification._Container,
      notification._Icon,
      notification._Text
    ]);
    testRenderer.update();

    notification.title = 'New title';

    // assert changes from _updateLayout happened
    await notification._updateLayoutSpyPromise;

    expect(notification._Container.h).toBe(
      notification._contentHeight +
        notification._actionAreaHeight +
        notification.style.actionAreaMarginTop +
        notification._totalMarginY
    );
  });

  it('should update the layout if changes are made while the Notification is activated without an action area', async () => {
    [notification, testRenderer] = createNotification(
      {
        entered: true,
        title: 'Title',
        description: 'description'
      },
      { spyOnMethods: ['enter', '_updateLayout'] }
    );

    await notification._enterSpyPromise;
    TestUtils.fastForward([
      notification._Container,
      notification._Icon,
      notification._Text
    ]);
    testRenderer.update();

    notification.title = 'New title';

    // assert changes from _updateLayout happened
    await notification._updateLayoutSpyPromise;

    expect(notification._Container.h).toBe(
      notification._contentHeight + notification._totalMarginY
    );
  });

  it('should call enter() and complete transitions', async () => {
    await notification.enter();

    TestUtils.fastForward([
      notification._Container,
      notification._Icon,
      notification._Text
    ]);
    testRenderer.update();

    expect(notification._Container.alpha).toEqual(
      notification.style.enterAlpha
    );
    expect(notification._Container.scale).toEqual(
      notification.style.enterScale
    );
    expect(notification._Icon.alpha).toEqual(notification.style.enterAlpha);
    expect(notification._Icon.scale).toEqual(notification.style.enterScale);
    expect(notification._Text.alpha).toEqual(notification.style.enterAlpha);
  });

  it('should call dismiss() and complete transitions', async () => {
    [notification, testRenderer] = createNotification(
      { entered: true, actionArea: 'Action Area' },
      { spyOnMethods: ['enter'] }
    );
    jest.spyOn(notification, 'signal');
    jest.spyOn(notification, 'fireAncestors');
    const { enterAlpha, enterScale, dismissAlpha, dismissScale } =
      notification.style;

    await notification._enterSpyPromise;
    TestUtils.fastForward([
      notification._Container,
      notification._Icon,
      notification._Text
    ]);
    testRenderer.update();

    expect(notification._Container.alpha).toEqual(enterAlpha);
    expect(notification._Container.scale).toEqual(enterScale);
    expect(notification.signal).not.toHaveBeenCalledWith(
      'notificationDismissed'
    );
    expect(notification.fireAncestors).not.toHaveBeenCalledWith(
      '$notificationDismissed'
    );

    await notification.dismiss();
    TestUtils.fastForward([
      notification._Container,
      notification._Icon,
      notification._Text
    ]);
    testRenderer.update();

    expect(notification._Container.alpha).toEqual(dismissAlpha);
    expect(notification._Container.scale).toEqual(dismissScale);
    expect(notification._Icon.alpha).toEqual(dismissAlpha);
    expect(notification._Text.alpha).toEqual(dismissAlpha);
    expect(notification.signal).toHaveBeenCalledWith('notificationDismissed');
    expect(notification.fireAncestors).toHaveBeenCalledWith(
      '$notificationDismissed'
    );
  });

  it('should not run dismiss animations if the Nofication is not activated', async () => {
    jest.spyOn(notification, 'signal');
    jest.spyOn(notification, 'fireAncestors');

    expect(notification.signal).not.toHaveBeenCalled();
    expect(notification.fireAncestors).not.toHaveBeenCalled();

    await notification.dismiss();
    TestUtils.fastForward([
      notification._Container,
      notification._Icon,
      notification._Text
    ]);

    expect(notification.signal).not.toHaveBeenCalled();
    expect(notification.fireAncestors).not.toHaveBeenCalled();
  });

  describe('when a notification is displayed', () => {
    beforeEach(async () => {
      jest.spyOn(notification, 'signal');
      jest.spyOn(notification, 'fireAncestors');
      expect(notification.signal).not.toHaveBeenCalled();
      expect(notification.fireAncestors).not.toHaveBeenCalled();

      await notification.enter();
      expect(notification.signal).toHaveBeenCalledWith('notificationEntered');
      expect(notification.fireAncestors).toHaveBeenCalledWith(
        '$notificationEntered'
      );
      jest.clearAllMocks();
    });

    it('should call enter() and complete transitions', async () => {
      TestUtils.fastForward([
        notification._Container,
        notification._Icon,
        notification._Text
      ]);
      testRenderer.update();

      expect(notification._Container.alpha).toEqual(
        notification.style.enterAlpha
      );
      expect(notification._Container.scale).toEqual(
        notification.style.enterScale
      );
      expect(notification._Icon.alpha).toEqual(notification.style.enterAlpha);
      expect(notification._Icon.scale).toEqual(notification.style.enterScale);
      expect(notification._Text.alpha).toEqual(notification.style.enterAlpha);
    });

    it('should dismiss an open notification', async () => {
      expect(notification.signal).not.toHaveBeenCalled();
      expect(notification.fireAncestors).not.toHaveBeenCalled();

      await notification.dismiss();
      expect(notification.signal).toHaveBeenCalledWith('notificationDismissed');
      expect(notification.fireAncestors).toHaveBeenCalledWith(
        '$notificationDismissed'
      );
    });

    it('should not emit multiple signals if enter is invoked after the notifcation is already open', async () => {
      expect(notification.signal).not.toHaveBeenCalled();
      expect(notification.fireAncestors).not.toHaveBeenCalled();

      await notification.enter();
      expect(notification.signal).not.toHaveBeenCalled();
      expect(notification.fireAncestors).not.toHaveBeenCalled();
    });
  });
});
