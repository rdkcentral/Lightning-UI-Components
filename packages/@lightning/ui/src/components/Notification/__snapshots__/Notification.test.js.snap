// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders an active Notification 1`] = `
{
  "Component": {
    "Container": {
      "active": true,
      "alpha": 1,
      "attached": true,
      "boundsMargin": null,
      "children": {
        "ActionArea": {
          "active": true,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "children": {
            "Text": {
              "active": true,
              "alpha": 1,
              "attached": true,
              "boundsMargin": null,
              "children": {
                "Text": {
                  "active": true,
                  "alpha": 1,
                  "attached": true,
                  "boundsMargin": null,
                  "clipping": false,
                  "color": 4294967295,
                  "enabled": true,
                  "flex": false,
                  "flexItem": false,
                  "h": 0,
                  "isComponent": undefined,
                  "mount": 0,
                  "mountX": 0,
                  "mountY": 0,
                  "pivot": 0.5,
                  "pivotX": 0.5,
                  "pivotY": 0.5,
                  "ref": "Text",
                  "renderOfScreen": undefined,
                  "renderToTexture": false,
                  "scale": 1,
                  "scaleX": 1,
                  "scaleY": 1,
                  "state": undefined,
                  "tag": [Function],
                  "tags": [
                    "Text",
                  ],
                  "texture": {
                    "fontFace": "Arial",
                    "fontSize": 20,
                    "lineHeight": 32,
                    "maxLines": 1,
                    "text": "action area",
                    "textBaseline": "bottom",
                    "textColor": 4294506490,
                    "type": "TextTexture",
                    "verticalAlign": "middle",
                    "wordWrapWidth": 576,
                  },
                  "visible": true,
                  "w": 0,
                  "x": 0,
                  "y": 2,
                  "zIndex": 0,
                },
              },
              "clipping": false,
              "color": 4294967295,
              "enabled": true,
              "flex": false,
              "flexItem": false,
              "h": 32,
              "hasFinalFocus": false,
              "hasFocus": false,
              "isComponent": true,
              "mount": 0,
              "mountX": 0,
              "mountY": 0,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Text",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": "",
              "tag": [Function],
              "tags": [
                "Text",
              ],
              "type": "TextBox",
              "visible": true,
              "w": 11,
              "x": 0,
              "y": 0,
              "zIndex": 0,
            },
          },
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": false,
          "flexItem": false,
          "h": 32,
          "isComponent": undefined,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "ActionArea",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": [
            "ActionArea",
          ],
          "visible": true,
          "w": 596,
          "x": 30,
          "y": 134,
          "zIndex": 0,
        },
        "Background": {
          "active": true,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": false,
          "flexItem": false,
          "h": 0,
          "isComponent": undefined,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Background",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": [
            "Background",
          ],
          "texture": {
            "type": "StaticCanvasTexture",
          },
          "visible": true,
          "w": 0,
          "x": 0,
          "y": 0,
          "zIndex": 0,
        },
        "Content": {
          "active": true,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "children": {
            "Icon": {
              "active": true,
              "alpha": 1,
              "attached": true,
              "boundsMargin": null,
              "clipping": false,
              "color": 4294967295,
              "enabled": true,
              "flex": false,
              "flexItem": true,
              "h": 80,
              "hasFinalFocus": false,
              "hasFocus": false,
              "isComponent": true,
              "mount": 0,
              "mountX": 0,
              "mountY": 0,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Icon",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": "",
              "tag": [Function],
              "tags": [
                "Icon",
              ],
              "type": "Icon",
              "visible": true,
              "w": 80,
              "x": 0,
              "y": 0,
              "zIndex": 0,
            },
            "Text": {
              "active": true,
              "alpha": 1,
              "attached": true,
              "boundsMargin": null,
              "children": {
                "Description": {
                  "active": true,
                  "alpha": 1,
                  "attached": true,
                  "boundsMargin": null,
                  "children": {
                    "Text": {
                      "active": true,
                      "alpha": 1,
                      "attached": true,
                      "boundsMargin": null,
                      "clipping": false,
                      "color": 4294967295,
                      "enabled": true,
                      "flex": false,
                      "flexItem": false,
                      "h": 0,
                      "isComponent": undefined,
                      "mount": 0,
                      "mountX": 0,
                      "mountY": 0,
                      "pivot": 0.5,
                      "pivotX": 0.5,
                      "pivotY": 0.5,
                      "ref": "Text",
                      "renderOfScreen": undefined,
                      "renderToTexture": false,
                      "scale": 1,
                      "scaleX": 1,
                      "scaleY": 1,
                      "state": undefined,
                      "tag": [Function],
                      "tags": [
                        "Text",
                      ],
                      "texture": {
                        "fontFace": "Arial",
                        "fontSize": 20,
                        "lineHeight": 32,
                        "text": "Description",
                        "textBaseline": "bottom",
                        "textColor": 3019438074,
                        "type": "TextTexture",
                        "verticalAlign": "middle",
                        "wordWrapWidth": 476,
                      },
                      "visible": true,
                      "w": 0,
                      "x": 0,
                      "y": 2,
                      "zIndex": 0,
                    },
                  },
                  "clipping": false,
                  "color": 4294967295,
                  "enabled": true,
                  "flex": false,
                  "flexItem": true,
                  "h": 32,
                  "hasFinalFocus": false,
                  "hasFocus": false,
                  "isComponent": true,
                  "mount": 0,
                  "mountX": 0,
                  "mountY": 0,
                  "pivot": 0.5,
                  "pivotX": 0.5,
                  "pivotY": 0.5,
                  "ref": "Description",
                  "renderOfScreen": undefined,
                  "renderToTexture": false,
                  "scale": 1,
                  "scaleX": 1,
                  "scaleY": 1,
                  "state": "",
                  "tag": [Function],
                  "tags": [
                    "Description",
                  ],
                  "type": "TextBox",
                  "visible": true,
                  "w": 11,
                  "x": 0,
                  "y": 0,
                  "zIndex": 0,
                },
                "Title": {
                  "active": true,
                  "alpha": 1,
                  "attached": true,
                  "boundsMargin": null,
                  "children": {
                    "Text": {
                      "active": true,
                      "alpha": 1,
                      "attached": true,
                      "boundsMargin": null,
                      "clipping": false,
                      "color": 4294967295,
                      "enabled": true,
                      "flex": false,
                      "flexItem": false,
                      "h": 0,
                      "isComponent": undefined,
                      "mount": 0,
                      "mountX": 0,
                      "mountY": 0,
                      "pivot": 0.5,
                      "pivotX": 0.5,
                      "pivotY": 0.5,
                      "ref": "Text",
                      "renderOfScreen": undefined,
                      "renderToTexture": false,
                      "scale": 1,
                      "scaleX": 1,
                      "scaleY": 1,
                      "state": undefined,
                      "tag": [Function],
                      "tags": [
                        "Text",
                      ],
                      "texture": {
                        "fontFace": "Arial",
                        "fontSize": 25,
                        "lineHeight": 36,
                        "text": "Title",
                        "textBaseline": "bottom",
                        "textColor": 4294506490,
                        "type": "TextTexture",
                        "verticalAlign": "middle",
                        "wordWrapWidth": 476,
                      },
                      "visible": true,
                      "w": 0,
                      "x": 0,
                      "y": 2,
                      "zIndex": 0,
                    },
                  },
                  "clipping": false,
                  "color": 4294967295,
                  "enabled": true,
                  "flex": false,
                  "flexItem": true,
                  "h": 36,
                  "hasFinalFocus": false,
                  "hasFocus": false,
                  "isComponent": true,
                  "mount": 0,
                  "mountX": 0,
                  "mountY": 0,
                  "pivot": 0.5,
                  "pivotX": 0.5,
                  "pivotY": 0.5,
                  "ref": "Title",
                  "renderOfScreen": undefined,
                  "renderToTexture": false,
                  "scale": 1,
                  "scaleX": 1,
                  "scaleY": 1,
                  "state": "",
                  "tag": [Function],
                  "tags": [
                    "Title",
                  ],
                  "type": "TextBox",
                  "visible": true,
                  "w": 5,
                  "x": 0,
                  "y": 0,
                  "zIndex": 0,
                },
              },
              "clipping": false,
              "color": 4294967295,
              "enabled": true,
              "flex": true,
              "flexItem": true,
              "h": 0,
              "isComponent": undefined,
              "mount": 0,
              "mountX": 0,
              "mountY": 0,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Text",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": undefined,
              "tag": [Function],
              "tags": [
                "Text",
              ],
              "visible": true,
              "w": 0,
              "x": 0,
              "y": 0,
              "zIndex": 0,
            },
          },
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": true,
          "flexItem": true,
          "h": 88,
          "isComponent": undefined,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Content",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": [
            "Content",
          ],
          "visible": true,
          "w": 596,
          "x": 30,
          "y": 16,
          "zIndex": 0,
        },
      },
      "clipping": true,
      "color": 4294967295,
      "enabled": true,
      "flex": false,
      "flexItem": false,
      "h": 182,
      "hasFinalFocus": false,
      "hasFocus": false,
      "isComponent": true,
      "mount": 0,
      "mountX": 0,
      "mountY": 0,
      "pivot": 0.5,
      "pivotX": 0.5,
      "pivotY": 0.5,
      "ref": "Container",
      "renderOfScreen": undefined,
      "renderToTexture": false,
      "scale": 1,
      "scaleX": 1,
      "scaleY": 1,
      "state": "",
      "tag": [Function],
      "tags": [
        "Container",
      ],
      "type": "Surface",
      "visible": true,
      "w": 636,
      "x": 0,
      "y": 0,
      "zIndex": 0,
    },
  },
}
`;

exports[`Notification renders an inactive Notification 1`] = `
{
  "Component": {
    "Container": {
      "active": false,
      "alpha": 0.001,
      "attached": true,
      "boundsMargin": null,
      "children": {
        "ActionArea": {
          "active": false,
          "alpha": 0.001,
          "attached": true,
          "boundsMargin": null,
          "children": {
            "Text": {
              "active": false,
              "alpha": 0.001,
              "attached": true,
              "boundsMargin": null,
              "clipping": false,
              "color": 4294967295,
              "enabled": true,
              "flex": false,
              "flexItem": false,
              "h": 0,
              "hasFinalFocus": false,
              "hasFocus": false,
              "isComponent": true,
              "mount": 0,
              "mountX": 0,
              "mountY": 0,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Text",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": "",
              "tag": [Function],
              "tags": [
                "Text",
              ],
              "type": "TextBox",
              "visible": true,
              "w": 0,
              "x": 0,
              "y": 0,
              "zIndex": 0,
            },
          },
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": false,
          "flexItem": false,
          "h": 0,
          "isComponent": undefined,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "ActionArea",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": [
            "ActionArea",
          ],
          "visible": true,
          "w": 0,
          "x": 0,
          "y": 0,
          "zIndex": 0,
        },
        "Background": {
          "active": false,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": false,
          "flexItem": false,
          "h": 0,
          "isComponent": undefined,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Background",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": [
            "Background",
          ],
          "texture": {
            "type": "StaticCanvasTexture",
          },
          "visible": true,
          "w": 0,
          "x": 0,
          "y": 0,
          "zIndex": 0,
        },
        "Content": {
          "active": false,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "children": {
            "Icon": {
              "active": false,
              "alpha": 0.001,
              "attached": true,
              "boundsMargin": null,
              "clipping": false,
              "color": 4294967295,
              "enabled": true,
              "flex": false,
              "flexItem": true,
              "h": 80,
              "hasFinalFocus": false,
              "hasFocus": false,
              "isComponent": true,
              "mount": 0,
              "mountX": 0,
              "mountY": 0,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Icon",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": "",
              "tag": [Function],
              "tags": [
                "Icon",
              ],
              "texture": {
                "src": "patch/to/src",
                "type": "ImageTexture",
              },
              "type": "Icon",
              "visible": true,
              "w": 80,
              "x": 0,
              "y": 0,
              "zIndex": 0,
            },
            "Text": {
              "active": false,
              "alpha": 0.001,
              "attached": true,
              "boundsMargin": null,
              "children": {
                "Description": {
                  "active": false,
                  "alpha": 0.001,
                  "attached": true,
                  "boundsMargin": null,
                  "children": {
                    "Text": {
                      "active": false,
                      "alpha": 1,
                      "attached": true,
                      "boundsMargin": null,
                      "clipping": false,
                      "color": 4294967295,
                      "enabled": true,
                      "flex": false,
                      "flexItem": false,
                      "h": 0,
                      "isComponent": undefined,
                      "mount": 0,
                      "mountX": 0,
                      "mountY": 0,
                      "pivot": 0.5,
                      "pivotX": 0.5,
                      "pivotY": 0.5,
                      "ref": "Text",
                      "renderOfScreen": undefined,
                      "renderToTexture": false,
                      "scale": 1,
                      "scaleX": 1,
                      "scaleY": 1,
                      "state": undefined,
                      "tag": [Function],
                      "tags": [
                        "Text",
                      ],
                      "texture": {
                        "fontFace": "Arial",
                        "fontSize": 20,
                        "lineHeight": 32,
                        "text": "This is a description",
                        "textBaseline": "bottom",
                        "textColor": 3019438074,
                        "type": "TextTexture",
                        "verticalAlign": "middle",
                        "wordWrapWidth": 476,
                      },
                      "visible": true,
                      "w": 0,
                      "x": 0,
                      "y": 2,
                      "zIndex": 0,
                    },
                  },
                  "clipping": false,
                  "color": 4294967295,
                  "enabled": true,
                  "flex": false,
                  "flexItem": true,
                  "h": 0,
                  "hasFinalFocus": false,
                  "hasFocus": false,
                  "isComponent": true,
                  "mount": 0,
                  "mountX": 0,
                  "mountY": 0,
                  "pivot": 0.5,
                  "pivotX": 0.5,
                  "pivotY": 0.5,
                  "ref": "Description",
                  "renderOfScreen": undefined,
                  "renderToTexture": false,
                  "scale": 1,
                  "scaleX": 1,
                  "scaleY": 1,
                  "state": "",
                  "tag": [Function],
                  "tags": [
                    "Description",
                  ],
                  "type": "TextBox",
                  "visible": true,
                  "w": 0,
                  "x": 0,
                  "y": 0,
                  "zIndex": 0,
                },
                "Title": {
                  "active": false,
                  "alpha": 0.001,
                  "attached": true,
                  "boundsMargin": null,
                  "children": {
                    "Text": {
                      "active": false,
                      "alpha": 1,
                      "attached": true,
                      "boundsMargin": null,
                      "clipping": false,
                      "color": 4294967295,
                      "enabled": true,
                      "flex": false,
                      "flexItem": false,
                      "h": 0,
                      "isComponent": undefined,
                      "mount": 0,
                      "mountX": 0,
                      "mountY": 0,
                      "pivot": 0.5,
                      "pivotX": 0.5,
                      "pivotY": 0.5,
                      "ref": "Text",
                      "renderOfScreen": undefined,
                      "renderToTexture": false,
                      "scale": 1,
                      "scaleX": 1,
                      "scaleY": 1,
                      "state": undefined,
                      "tag": [Function],
                      "tags": [
                        "Text",
                      ],
                      "texture": {
                        "fontFace": "Arial",
                        "fontSize": 25,
                        "lineHeight": 36,
                        "text": "Title",
                        "textBaseline": "bottom",
                        "textColor": 4294506490,
                        "type": "TextTexture",
                        "verticalAlign": "middle",
                        "wordWrapWidth": 476,
                      },
                      "visible": true,
                      "w": 0,
                      "x": 0,
                      "y": 2,
                      "zIndex": 0,
                    },
                  },
                  "clipping": false,
                  "color": 4294967295,
                  "enabled": true,
                  "flex": false,
                  "flexItem": true,
                  "h": 0,
                  "hasFinalFocus": false,
                  "hasFocus": false,
                  "isComponent": true,
                  "mount": 0,
                  "mountX": 0,
                  "mountY": 0,
                  "pivot": 0.5,
                  "pivotX": 0.5,
                  "pivotY": 0.5,
                  "ref": "Title",
                  "renderOfScreen": undefined,
                  "renderToTexture": false,
                  "scale": 1,
                  "scaleX": 1,
                  "scaleY": 1,
                  "state": "",
                  "tag": [Function],
                  "tags": [
                    "Title",
                  ],
                  "type": "TextBox",
                  "visible": true,
                  "w": 0,
                  "x": 0,
                  "y": 0,
                  "zIndex": 0,
                },
              },
              "clipping": false,
              "color": 4294967295,
              "enabled": true,
              "flex": true,
              "flexItem": true,
              "h": 0,
              "isComponent": undefined,
              "mount": 0,
              "mountX": 0,
              "mountY": 0,
              "pivot": 0.5,
              "pivotX": 0.5,
              "pivotY": 0.5,
              "ref": "Text",
              "renderOfScreen": undefined,
              "renderToTexture": false,
              "scale": 1,
              "scaleX": 1,
              "scaleY": 1,
              "state": undefined,
              "tag": [Function],
              "tags": [
                "Text",
              ],
              "visible": true,
              "w": 0,
              "x": 0,
              "y": 0,
              "zIndex": 0,
            },
          },
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": true,
          "flexItem": true,
          "h": 8,
          "isComponent": undefined,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Content",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": [
            "Content",
          ],
          "visible": true,
          "w": 596,
          "x": 20,
          "y": 16,
          "zIndex": 0,
        },
      },
      "clipping": true,
      "color": 4294967295,
      "enabled": true,
      "flex": false,
      "flexItem": false,
      "h": 120,
      "hasFinalFocus": false,
      "hasFocus": false,
      "isComponent": true,
      "mount": 0,
      "mountX": 0,
      "mountY": 0,
      "pivot": 0.5,
      "pivotX": 0.5,
      "pivotY": 0.5,
      "ref": "Container",
      "renderOfScreen": undefined,
      "renderToTexture": false,
      "scale": 1,
      "scaleX": 1,
      "scaleY": 1,
      "state": "",
      "tag": [Function],
      "tags": [
        "Container",
      ],
      "type": "Surface",
      "visible": true,
      "w": 120,
      "x": 516,
      "y": 0,
      "zIndex": 0,
    },
  },
}
`;
